cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/deploy/pipeline/SystemMetadataWriterStageTest.java

```diff
@@ -92,19 +94,21 @@ public void testWorkflowTags() throws Exception {
     systemMetadataWriterStage.process(appWithPrograms);
 
     // verify that the workflow is not tagged with the fork node name. First wait for tags to show up
-    Tasks.waitFor(false, () -> metadataStore.getTags(
-      MetadataScope.SYSTEM, appId.workflow(workflowName).toMetadataEntity()).isEmpty(), 5, TimeUnit.SECONDS);
-    Set<String> workflowSystemTags = metadataStore.getTags(MetadataScope.SYSTEM,
-                                                           appId.workflow(workflowName).toMetadataEntity());
+    Tasks.waitFor(false, () -> metadataStorage.read(new Read(appId.workflow(workflowName).toMetadataEntity(),
+                                                             MetadataScope.SYSTEM, MetadataKind.TAG)).isEmpty(),
+                  5, TimeUnit.SECONDS);
+    Set<String> workflowSystemTags = metadataStorage
+      .read(new Read(appId.workflow(workflowName).toMetadataEntity())).getTags(MetadataScope.SYSTEM);
     Sets.SetView<String> intersection = Sets.intersection(workflowSystemTags, getWorkflowForkNodes(workflowSpec));
     Assert.assertTrue("Workflows should not be tagged with fork node names, but found the following fork nodes " +
                         "in the workflow's system tags: " + intersection, intersection.isEmpty());
 
     // verify that metadata was added for the workflow's schedule. First wait for the metadata to show up
-    Tasks.waitFor(false, () -> metadataStore.getProperties(
-      MetadataScope.SYSTEM, appId.toMetadataEntity()).isEmpty(), 5, TimeUnit.SECONDS);
-    Map<String, String> metadataProperties = metadataStore.getMetadata(MetadataScope.SYSTEM,
-                                                                       appId.toMetadataEntity()).getProperties();
+    Tasks.waitFor(false, () -> metadataStorage.read(new Read(appId.toMetadataEntity(),
+                                                             MetadataScope.SYSTEM, MetadataKind.TAG)).isEmpty(),
```

Comment:
Typo: the old code is reading properties not tags

Reply:
oops. will fix.